File: boj/b_13398.js
function solution(n, array) {
  const [tableA, tableB] = [[...array], [...array]];
  for (let i = 1; i < n; i += 1) {
    tableA[i] = Math.max(tableA[i], tableA[i] + tableA[i - 1]);
    tableB[i] = Math.max(tableA[i - 1], tableB[i] + tableB[i - 1]);
  }
  const [a, b] = [Math.max(...tableA), Math.max(...tableB)];
  console.log(a, b);
  return a > b ? a : b;
}
console.log(solution(10, [10, -4, 3, 1, 5, 6, -35, 12, 21, -1]));

File: boj/b_13398.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
table_a = list(map(int, input().split()))
table_b = [_ for _ in table_a]
for i in range(1, n):
    table_a[i] = max(table_a[i], table_a[i] + table_a[i - 1])
    table_b[i] = max(table_a[i - 1], table_b[i] + table_b[i - 1])
a, b = max(table_a), max(table_b)
print(a if a > b else b)